sktime/transformations/series/impute.py

```diff
@@ -0,0 +1,150 @@
+        super(Imputer, self).__init__()
+
+    def transform(self, Z, X=None):
+        """Transform data.
+        Returns a transformed version of Z.
+
+        Parameters
+        ----------
+        Z : pd.Series
+
+        Returns
+        -------
+        z : pd.Series
+            Transformed time series.
+        """
+        self.check_is_fitted()
+        self._check_method()
+        z = check_series(Z, enforce_univariate=True)
+
+        # replace missing_values with np.nan
+        if self.missing_values:
+            z = z.replace(to_replace=self.missing_values, value=np.nan)
+
+        if self.method == "random":
+            z = z.apply(lambda x: self._get_random(z) if np.isnan(x) else x)
+        elif self.method == "constant":
+            z = z.fillna(value=self.value)
+        elif self.method in ["backfill", "bfill", "pad", "ffill"]:
+            z = z.fillna(method=self.method)
+        elif self.method in ["drift", "forecaster"]:
+            if self.method == "forecaster":
+                forecaster = self.forecaster
+            else:
+                forecaster = PolynomialTrendForecaster(degree=1)
+            # in-sample forecasting horizon
+            fh_ins = -np.arange(len(z))
+            # fill NaN before fitting with ffill and backfill (heuristic)
+            z_pred = forecaster.fit(
+                z.fillna(method="ffill").fillna(method="backfill")
+            ).predict(fh=fh_ins)
+            # fill with trend values
+            z = z.fillna(value=z_pred)
+        elif self.method == "mean":
+            z = z.fillna(value=z.mean())
+        elif self.method == "median":
+            z = z.fillna(value=z.median())
+        elif self.method in ["nearest", "linear"]:
+            z = z.interpolate(method=self.method)
+        else:
+            raise ValueError(f"method {self.method} not available")
+        return pd.Series(z)
+
+    def _check_method(self):
+        if (
+            self.value is not None
+            and self.method != "constant"
+            or self.method == "constant"
+            and self.value is None
+        ):
+            raise ValueError(
+                """Imputing with a value can only be
+                used if method=\"value\" and if arg value is not None"""
+            )
+        elif (
+            self.forecaster is not None
+            and self.method != "forecaster"
+            or self.method == "forecaster"
+            and self.forecaster is None
+        ):
+            raise ValueError(
+                """Imputing with a forecaster can only be used if
+                method=\"forecaster\" and if arg forecaster is not None"""
+            )
+        else:
+            pass
+
+    def _get_random(self, z):
+        """Create a random int or float value
+
+        :param z: Series
+        :type z: pd.Series
+        :return: Random int or float between min and max of z
+        :rtype: int/float
+        """
+        random.seed(self.random_state)
```

Comment:
Can we use `check_random_state` from scikit-learn and numpy's random state object here? So something like this works: 
```python
rng = check_random_state(self.random_state)
rng.randint(z.min(), z.max())
```

Reply:
@mloning thx, this is much better  👍🏻 ! Because otherwise doing `method="random"` would impute all `np.nan` with the same value. So now, if user sets seed, there are different values imputed